controllers/hetznercluster_controller_test.go

```diff
@@ -844,55 +843,71 @@ var _ = Describe("reconcileRateLimit", func() {
 })
 
 func TestSetControlPlaneEndpoint(t *testing.T) {
-	t.Run("function should return false if hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled is not enabled and hetznerCluster.Spec.ControlPlaneEndpoint is nil", func(t *testing.T) {
-		hetznerCluster := &infrav1.HetznerCluster{Spec: infrav1.HetznerClusterSpec{ControlPlaneLoadBalancer: infrav1.LoadBalancerSpec{Enabled: false}, ControlPlaneEndpoint: nil}}
-		got := SetControlPlaneEndpoint(hetznerCluster)
-		if (hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled == true){
-			t.Fatalf("Wrong value for ControlPlaneLoadBalancer.Enabled. Got: %v, Want: false", hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled)
+	t.Run("return false and don't make changes to ControlPlaneEndpoint if load balancer is not enabled and ControlPlaneEndpoint is nil", func(t *testing.T) {
+		hetznerCluster := &infrav1.HetznerCluster{
+			Spec: infrav1.HetznerClusterSpec{
+				ControlPlaneLoadBalancer: infrav1.LoadBalancerSpec{Enabled: false}, 
+				ControlPlaneEndpoint: nil},
 		}
+		got := SetControlPlaneEndpoint(hetznerCluster)
 		if (hetznerCluster.Spec.ControlPlaneEndpoint != nil){
 			t.Fatalf("Wrong value for hetznerCluster.Spec.ControlPlaneEndpoint. Got: %s, Want: <nil>", hetznerCluster.Spec.ControlPlaneEndpoint)
 		}
-		assert.Equal(t, false, got)
+		if got != false {
+			t.Fatal("return value should be false")
+		}
 	})
-	t.Run("function should return true if hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled is false but hetznerCluster.Spec.ControlPlaneEndpoint is not nil", func(t *testing.T) {
-		hetznerCluster := &infrav1.HetznerCluster{Spec: infrav1.HetznerClusterSpec{ControlPlaneLoadBalancer: infrav1.LoadBalancerSpec{Enabled: false}, ControlPlaneEndpoint: &clusterv1.APIEndpoint{Host: "xyz"}}}
-		got := SetControlPlaneEndpoint(hetznerCluster)
-		if (hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled != false){
-			t.Fatalf("Wrong value for ControlPlaneLoadBalancer.Enabled. Got: %v, Want: false", hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled)
+
+	t.Run("return true and don't make changes to ControlPlaneEndpoint if load balancer is not enabled but ControlPlaneEndpoint is not nil", func(t *testing.T) {
+		hetznerCluster := &infrav1.HetznerCluster{
+			Spec: infrav1.HetznerClusterSpec{
+				ControlPlaneLoadBalancer: infrav1.LoadBalancerSpec{Enabled: false}, 
+				ControlPlaneEndpoint: &clusterv1.APIEndpoint{Host: "xyz"}},
```

Comment:
you don't have a check for the port

Reply:
I had it this way intentionally becuase the requirement is only to not have ControlPlaneEndpoint not nil. So I thought keeping it this way will allow it to pass through.

Reply:
In this test case you wanted to check that ControlPlaneEndpoint does not change. You had left the port empty before. I believe that was better as now, where you added the port to ControlPlaneEndpoint as well. 

Leave it empty and check in the test that it is left empty!

Reply:
Okay I'm a little confused with this one @janiskemper. How do I leave an integer input field empty? Do you mean I should set its value to 0, or i just leave it unmnetioned? If i set it to zero its value will not change since we do not enter that condition in this test. I'm sorry I'm just a little confused

Reply:
0

Reply:
and yes, you should test that the 0 is not replaced by anything